01-Authorization/auth0authorization/views.py

```diff
@@ -0,0 +1,51 @@
+from functools import wraps
+
+from django.http import HttpResponse
+from rest_framework.decorators import api_view
+from jose import jwt
+
+# Create your views here.
+
+
+def get_token_auth_header(request):
+    """Obtains the access token from the Authorization Header
+    """
+    auth = request.META.get("HTTP_AUTHORIZATION", None)
+    parts = auth.split()
+    token = parts[1]
+
+    return token
+
+
+def requires_scope(required_scope):
+    """Determines if the required scope is present in the access token
+    Args:
+        required_scope (str): The scope required to access the resource
+    """
+    def require_scope(f):
+        @wraps(f)
+        def decorated(*args, **kwargs):
+            token = get_token_auth_header(args[0])
+            unverified_claims = jwt.get_unverified_claims(token)
+            token_scopes = unverified_claims["scope"].split()
```

Comment:
We are asking for unverified claims without validating the token. 

Shouldn't we be calling http://python-jose.readthedocs.io/en/latest/jwt/api.html#jose.jwt.decode?

Reply:
The decorator `@api_view` checks for token validity. To ask for scopes, `@api_view` and `@requires_scope` should be used.